compute/ARMComputeEx/src/core/CL/kernels/CLSplitVKernelEx.cpp

```diff
@@ -0,0 +1,253 @@
+{
+  unsigned int axis_offset = 0;
+  // Start/End coordinates
+  Coordinates start_coords;
+  Coordinates end_coords;
+  for (unsigned int d = 0; d < input->info()->num_dimensions(); ++d)
+  {
+    end_coords.set(d, -1);
+  }
+  int out_iter = 0;
+  for (const auto &output : outputs)
+  {
+    const TensorShape output_shape = output->info()->tensor_shape();
+    auto op_size = output_shape.total_size();
+    if (!op_size)
+    {
+      continue;
+    }
+
+    assert(op_size != 0);
+    assert(split_dim_revised <= output_shape.num_dimensions());
+
+    const size_t axis_split_step = output_shape[split_dim_revised];
+
+    // Output auto inizialitation if not yet initialized
+    TensorInfo tmp_output_info = *output->info()->clone();
+    auto_init_if_empty(
+        tmp_output_info,
+        input->info()->clone()->set_is_resizable(true).set_tensor_shape(output_shape));
+
+    // Update coordinate on axis
+    start_coords.set(split_dim_revised, axis_offset);
+    end_coords.set(split_dim_revised, axis_offset + axis_split_step);
+
+    // Configure slice function
+    _slice_functions[out_iter].configure(input, output, start_coords, end_coords);
+
+    // Set valid region from shape
+    outputs[out_iter++]->info()->set_valid_region(ValidRegion(Coordinates(), output_shape));
+    axis_offset += axis_split_step;
+  }
+}
+
+} // namespace
+
+CLSplitVKernel::CLSplitVKernel()
+    : _input(nullptr), _size_splits(nullptr), _outputs(), _split_dim(nullptr), _num_splits(0)
+{
+}
+
+void CLSplitVKernel::configure(const ICLTensor *input, const ICLTensor *size_splits,
+                               const ICLTensor *split_dim, const std::vector<ICLTensor *> &outputs,
+                               unsigned int num_splits, int layout_diff)
+{
+  ARM_COMPUTE_ERROR_ON_NULLPTR(input, size_splits);
+  ARM_COMPUTE_ERROR_THROW_ON(validate_arguments(size_splits, split_dim, outputs, num_splits));
+
+  _input = input;
+  _size_splits = size_splits;
+  _outputs = outputs;
+  _split_dim = split_dim;
+  _num_splits = num_splits;
+  _layout_diff = layout_diff;
+}
+
+void CLSplitVKernel::run()
+{
+  cl::CommandQueue &q = CLScheduler::get().queue();
+  const_cast<ICLTensor *>(_split_dim)->map(q);
+  // Fetch the split_dim value
+  int32_t split_dim = *(reinterpret_cast<int32_t *>(_split_dim->buffer()));
+  const_cast<ICLTensor *>(_split_dim)->unmap(q);
+
+  split_dim =
+      (split_dim < 0) ? (split_dim + _input->info()->tensor_shape().num_dimensions()) : split_dim;
+  uint32_t split_dim_revised = recompute_split_dim(_input, &split_dim, _layout_diff);
+
+  // Create tensor slices
+  std::vector<CLSlice> _slice_functions;
+  _slice_functions.resize(_num_splits);
+
+  // Extract output tensor info
+  std::vector<ITensorInfo *> outputs_info;
+  for (auto &output : _outputs)
+  {
+    ARM_COMPUTE_ERROR_ON_NULLPTR(output);
+    outputs_info.emplace_back(output->info());
+  }
+
+  // Validate slices
+  validate_slices(_input->info(), outputs_info, split_dim_revised);
```

Comment:
Could you wrap this in `ARM_COMPUTE_RETURN_ON_ERROR`?

Reply:
will replace with `ARM_COMPUTE_ERROR_THROW_ON(validate_slices(..));`